ckanext/dcat/logic.py

```diff
@@ -144,18 +144,24 @@ def _pagination_info(query, data_dict):
 
     def _page_url(page):
 
+        base_url = config.get('ckan.site_url', '').strip('/')
+        if not base_url:
+            base_url = toolkit.request.host_url
+        base_url = '%s%s%s' % (
+            base_url, toolkit.request.path_info, toolkit.request.script_name)
```

Comment:
`request.host_url` works on both, but `request.path_info` and `request.script_name` are not present on the Flask request object. Both Flask's and Pylons's have `request.path` though, which is the combination of both (eg `/feed/catalog.xml` for http://example.com/feed/catalog.xml`). Let's use that and future-proof this! :rocket: